lib/actions/attachmentToJson.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const { promisify } = require('util');
+const request = promisify(require('request'));
+
+const path = require('path');
+const xml2Json = require('../xml2Json.js');
+const fs = require('fs');
+
+
+function streamToString(stream) {
+    const chunks = [];
+    return new Promise((resolve, reject) => {
+        stream.on('data', chunks.push);
+        stream.on('error', reject);
+        stream.on('end', () => resolve(Buffer.concat(chunks).toString('utf8')));
+    });
+}
+
+module.exports.process = async function processAction(msg, cfg) {
+    let xml = {};
+
+    console.log('attachment to string started, message=%j cfg=%j', msg, cfg);
+
+    const attachments = msg.attachments;
+
+    console.log('Found %s attachments', Object.keys(attachments || {}).length);
+    for (let key in attachments || {}) {
+        const attachment = attachments[key];
+        const fileName = key;
+        console.log('Processing attachment=%s content=%j', fileName, attachment);
+
+        if (fileName.split('.').pop() === 'xml') {
+
+            const requestOptions = {
+                url: attachment.url,
+                method: 'get',
+                json: true,
+                body: '',
+                headers: ''
```

Comment:
If you aren't providing values for `body` or `headers`, you shouldn't include them on this object.